src/sanitiser.js

```diff
@@ -1,46 +1,82 @@
-import { globalConfig, testConfig } from './defaultConfig';
-
-const hasProperty = (obj, property) =>
-  Object.prototype.hasOwnProperty.call(obj, property);
-
-const sanitiseTestConfiguration = (conf) => {
-  const configuration = conf || {};
-  configuration.resolution = hasProperty(configuration, 'resolution')
-    ? configuration.resolution
-    : testConfig.resolution;
-  configuration.steps = hasProperty(configuration, 'steps')
-    ? configuration.steps
-    : testConfig.steps;
-  configuration.name = hasProperty(configuration, 'name')
-    ? configuration.name
-    : testConfig.name;
+import check from 'check-types';
+import type from 'type-detect';
+import logger from './utils/logger';
+import { globalConfig, testConfig } from './config/config';
+
+const error = (name, wrongType, correctType) =>
+  logger.error(`Invalid argument ${name} with type ${wrongType} been passed. Argument should be ${correctType}`);
+
+const checkProperty = (obj, property, checkType) => {
+  if (obj) {
```

Comment:
can this be refactored to collapse the two codepaths that return false?

Reply:
Done